### PR TITLE
fix: move user data to sub-folder and ensure folder existence

### LINKED ISSUE #98: Group "user data" files in subfolders
Currently we use `~/.config/yafti-last-run`.

Best practices is to create sub-folders, as such:

`~/.config/yafti/last-run` and `~/.local/share/yafti` and `~/.cache/yafti` etc.

Especially important if we decide to grow the amount of files we have per-user.

### PR DESCRIPTION
**Pull request by Arcitec**
Closes #98.

### CODE REVIEW

File: yafti/app.py [hunk 1]
```diff
@@ -16,6 +16,10 @@
 
 import hashlib
 
+## ---- TODO: START: REMOVE THIS HACK AFTER ENOUGH PEOPLE HAVE UPGRADED ----
+from pathlib import Path
+## ---- TODO: END: REMOVE THIS HACK AFTER ENOUGH PEOPLE HAVE UPGRADED ----
```
Comment (marcoceppi) [inline yafti/app.py:21]: These signposts are a bit aggressive. Instead I suggested how I'd like to see this tracked on the snippet below.

```suggestion
from pathlib import Path
```

File: yafti/app.py [hunk 2]
```diff
@@ -33,6 +37,15 @@ def __init__(self, cfg: Config = None, loop=None):
     def run(self, *args, **kwargs):
         configured_mode = self.config.properties.mode
         _p = self.config.properties.path.expanduser()
+        ## ---- TODO: START: REMOVE THIS HACK AFTER ENOUGH PEOPLE HAVE UPGRADED ----
+        _old_p = Path("~/.config/yafti-last-run").expanduser()
+        if _old_p.exists():
+            if not _p.parent.is_dir():
+                _p.parent.mkdir(mode=0o755, parents=True, exist_ok=True)
+            if _p.is_file():
```
Comment (marcoceppi) [inline yafti/app.py:45]: I don't think we want to remove the current properties path if it exists. Instead I'd opt to log a warning that both old path and new path were found but to defer to the new file being created.

File: yafti/app.py [hunk 3]
```diff
@@ -33,6 +37,15 @@ def __init__(self, cfg: Config = None, loop=None):
     def run(self, *args, **kwargs):
         configured_mode = self.config.properties.mode
         _p = self.config.properties.path.expanduser()
+        ## ---- TODO: START: REMOVE THIS HACK AFTER ENOUGH PEOPLE HAVE UPGRADED ----
```
Comment (marcoceppi) [inline yafti/app.py:40]:
```suggestion
        # TODO(GH-#): Remove this prior to 1.0 release
```

For todos that support some form of compatibility, I'd prefer we instead open a github issue describing what to remove and use that as `GH-#` where `#` is the issue id. That helps to provide more indepth context for those looking at the code about why this was added and when it should be removed. I think the 1.0 target is the most ideal for removal as it's our chance to break backwards compat and move forward. My guess is we're a few months away from 1.0 after the yafti.core rewrite and separation of runner logic from display logic.

**Review by marcoceppi — CHANGES_REQUESTED** [review]

File: yafti/app.py (hunk 3 above)
Comment (Arcitec) [reply yafti/app.py:40]: Yeah that's a good idea, I'm opening a ticket to track it and pointing the code comment at it now.

**Review by marcoceppi — APPROVED** [review]

**Comment by Arcitec** [comment]
@marcoceppi Just following up to see if that approve was meant to be a merge, since auto-merging after approval is disabled here. :)

**Pull request merged**